cdk/src/open-data-platform/data-plane/data-import/handler-factory.ts

```diff
@@ -1,44 +1,78 @@
 import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
-import { ConnectionPool } from '@databases/pg';
 import * as AWS from 'aws-sdk';
-import { connectToDb } from '../schema/schema.handler';
+import { AnyLengthString } from 'aws-sdk/clients/comprehendmedical';
 
 // These libraries don't have types, so they are imported in a different way.
 const { chain } = require('stream-chain');
 const { streamArray } = require('stream-json/streamers/StreamArray');
 const Batch = require('stream-json/utils/Batch');
 const Pick = require('stream-json/filters/Pick');
 
+export interface ImportRequest {
+  rowOffset?: number;
+  rowLimit?: number;
+  batchSize?: number;
+}
+
+interface ImportResult {
+  processedBatchCount: number;
+  sucessfulBatchCount: number;
+  erroredBatchCount: number;
+}
+
 const S3 = new AWS.S3();
 
 /**
  * Builds a lambda handler that parses a GeoJSON file and executes a callback.
+ *
+ * The callback takes in a set of rows and should insert them into the DB using the provided db
+ * instance.
+ *
+ * You can invoke the lambda via the web console and provide an ImportRequest to skip a set of rows
+ * or limit the import to a max number. The lambda will attempt to honor the limit, but it may
+ * exceed it due to how it proccesses rows in parallel. Example:
+ *
+ * {
+ *   "rowOffset": 0,
+ *   "rowLimit": 27000,
+ *   "batchSize": 10
+ * }
+ *
  * @param s3Params - Details for how to get the GeoJSON file.
  * @param callback - Function that operates on a single element from the file.
```

Comment:
nit: update wording to convey that the callback operates on a whole batch instead of a single row

Reply:
Done.